internal/controllers/opentelemetrycollector_controller.go

```diff
@@ -249,7 +249,7 @@ func (r *OpenTelemetryCollectorReconciler) Reconcile(ctx context.Context, req ct
 
 	// We have a deletion, short circuit and let the deletion happen
 	if deletionTimestamp := instance.GetDeletionTimestamp(); deletionTimestamp != nil {
-		if controllerutil.ContainsFinalizer(&instance, collectorFinalizer) {
+		if params.Config.CreateRBACPermissions == rbac.Available && controllerutil.ContainsFinalizer(&instance, collectorFinalizer) {
```

Comment:
We shouldn't have a condition here. If the CR has a finalizer, we should honor it.